hngprojects/hng_boilerplate_java_web
# feat: delete user by email

## How this should be manually tested
- Ensure the user is authenticated and authorized to delete an account.
- Access the route as an authenticated user `(ROLE_SUPER_ADMIN)`.
- Send a `DELETE` request to `/api/v1/users/delete` with the following request body to delete a user by their email:


#### Request
```
{
    "email": "user.email@example.com"
}
```

#### Responses
```
{
    "status_code": "success",
    "message": "The account has been successfully deleted."
}
```

```
{
    "status_code": "404",
    "message": "User not found with email: user.email@example.com"
}
```

```
{
    "status_code": "400",
    "message": "Bad Request. The email field is required."
}
```


## Checklist of What I Did:

- [ ] Implemented the `DELETE /api/v1/users/delete` endpoint to delete a user by email.
- [ ] Added validation to ensure the email field is provided and is not empty.
- [ ] Checked if the user exists by their email and handled cases where the user does not exist.
- [ ] Implemented error handling to return appropriate status codes and messages.
- [ ] Wrote unit tests to validate the service logic for various scenarios (successful deletion, user not found, bad request).


## Reference Issue
[https://github.com/hngprojects/hng_boilerplate_java_web/issues/417]()
